fs-index/src/index.rs

```diff
@@ -1,1120 +1,436 @@
-use anyhow::anyhow;
-use canonical_path::{CanonicalPath, CanonicalPathBuf};
-use itertools::Itertools;
-use std::collections::{HashMap, HashSet};
-use std::fs::{self, File, Metadata};
-use std::io::{BufRead, BufReader, Write};
-use std::ops::Add;
-use std::path::{Path, PathBuf};
-use std::time::{Duration, SystemTime, UNIX_EPOCH};
-use walkdir::{DirEntry, WalkDir};
+use std::{
+    collections::HashMap,
+    fs,
+    hash::Hash,
+    path::{Path, PathBuf},
+    time::{Duration, SystemTime},
+};
 
-use log;
+use serde::{Deserialize, Serialize};
 
-use data_error::{ArklibError, Result};
+use data_error::Result;
 use data_resource::ResourceId;
 use fs_storage::{ARK_FOLDER, INDEX_PATH};
 
-#[derive(Eq, Ord, PartialEq, PartialOrd, Hash, Clone, Debug)]
-pub struct IndexEntry<Id: ResourceId> {
-    pub modified: SystemTime,
-    pub id: Id,
-}
```

Comment:
This structure should be brought back, see the comment about invariant and `added: HashMap`

Reply:
I'm not sure where `IndexEntry` should be used. If it's for the `added` map in `IndexUpdate`, the user wouldn't have access to the paths of the added resources. I think we'd want this information to be available in our API.

Reply:
No, it was used in the `ResourceIndex` structure, in the `path_to_resource` part.